include/kodlab_mjbots_sdk/rotations.h

```diff
@@ -0,0 +1,487 @@
+   * @param rot_type_in rotation type (Extrinsic or Intrinsic)
+   */
+  EulerAngles(const Eigen::Vector3<Scalar> &angles_in,
+              const std::array<int, 3> &axes_in,
+              const EulerRotationType &rot_type_in)
+      : angles({angles_in(0), angles_in(1), angles_in(2)}), 
+        axes(axes_in),
+        rot_type(rot_type_in) {}
+
+  /**
+   * @brief Returns quaternion representation of this rotation.
+   *
+   * @return quaternion
+   */
+  Eigen::Quaternion<Scalar> ToQuaternion() const
+  {
+    return EulerAnglesToQuaternion(*this);
+  }
+
+  /**
+   * @brief Returns 3x3 rotation matrix representation of this rotation.
+   *
+   * @return 3x3 rotation matrix
+   */
+  Eigen::Matrix3<Scalar> ToRotationMatrix() const
+  {
+    return EulerAnglesToRotationMatrix(*this);
+  }
+
+  /**
+   * @brief Returns roll angle if using a [Tait-Bryan convention]
+   *        (https://en.wikipedia.org/wiki/Davenport_chained_rotations)
+   * @note Returned value is incorrect if convention is not Tait-Bryan (e.g. if 
+   *       using Pure Euler convention)
+   * 
+   * @return const Scalar 
+   */
+  const Scalar roll() const { return angles[tait_bryan_rpy_map_[0]]; }
+
+  /**
+   * @brief Returns pitch angle if using a [Tait-Bryan convention]
+   *        (https://en.wikipedia.org/wiki/Davenport_chained_rotations)
+   * @note Returned value is incorrect if convention is not Tait-Bryan (e.g. if 
+   *       using Pure Euler convention)
+   * 
+   * @return const Scalar 
+   */
+  const Scalar pitch() const { return angles[tait_bryan_rpy_map_[1]]; }
+
+  /**
+   * @brief Returns yaw angle if using a [Tait-Bryan convention]
+   *        (https://en.wikipedia.org/wiki/Davenport_chained_rotations)
+   * @note Returned value is incorrect if convention is not Tait-Bryan (e.g. if 
+   *       using Pure Euler convention)
+   * 
+   * @return const Scalar 
+   */
+  const Scalar yaw() const { return angles[tait_bryan_rpy_map_[2]]; }
+
+private:
+  /**
+   * @brief Indices map from axes convention to roll-pitch-yaw angles
+   */
+  std::array<int, 3> tait_bryan_rpy_map_ = {0, 1, 2};
+
+  /**
+   * @brief Computes indices map from axes convention to roll-pitch-yaw angles.
+   * @link https://en.wikipedia.org/wiki/Davenport_chained_rotations @endlink
+   * 
+   * @param axes array of Tait-Bryan a
+   */
+  void ComputeAngleIndexMap(const std::array<int, 3> &axes)
+  {
+    for (unsigned int i = 0; i < axes.size(); i++)
+    {
+      tait_bryan_rpy_map_[axes[i]] = i;
+    }
+  }
+
+  /**
+   * @brief Get the axes in extrinsic order
+   * 
+   * @return std::array<int,3> 
+   */
+  std::array<int,3> get_axes_extrinsic()
+  {
+    std::array<int,3> axes_out(axes);
+    if(rot_type==INTRINSIC){
+        std::reverse(axes_out.begin(),axes_out.end());
+    }
+    return axes_out;
+  }
+  
+  /**
+   * @brief Get the angles in extrinsic order
+   * 
+   * @return std::array<float,3> 
+   */
+  std::array<float,3> get_angles_extrinsic()
+  {
```

Comment:
This will not work as written, should read `std::array<float,3> angles_out(angles);`